graphics.ts

```diff
@@ -97,8 +171,159 @@ class Graphics {
      * @param h The height of the rectangle in pixels.
      */
     drawRect(x: number, y: number, w: number, h: number) {
+        this.ctx.strokeRect(x, y, w, h);
+    }
+
+    /**
+     * Fills a circle with the given radius at the given position.
+     * @param x The x coordinate for the center of the circle.
+     * @param y The y coordinate for the center of the circle.
+     * @param r The radius of the circle.
+     */
+    fillCircle(x: number, y: number, r:number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, 0, 2 * Math.PI);
+        this.ctx.fill(); 
+    }
+
+    /**
+     * Draws a circle with the given radius at the given position.
+     * @param x The x coordinate for the center of the circle.
+     * @param y The y coordinate for the center of the circle.
+     * @param r The radius of the circle.
+     */
+    drawCircle(x: number, y: number, r:number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, 0, 2 * Math.PI);
+        this.ctx.stroke(); 
+    }
+
+    /**
+     * Fills an arc at the given coordinates with the given radius and angles.
+     * @param x The x coordinate for the center of the arc.
+     * @param y The y coordinate for the center of the arc.
+     * @param r The radius for the arc.
+     * @param sa The start angle for the arc, in radians.
+     * @param ea The end angle for the arc, in radians.
+     */
+    fillArc(x: number, y: number, r:number, sa:number, ea: number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, sa, ea);
+        this.ctx.fill(); 
+    }
+
+    /**
+     * Draws an arc at the given coordinates with the given radius and angles.
+     * @param x The x coordinate for the center of the arc.
+     * @param y The y coordinate for the center of the arc.
+     * @param r The radius for the arc.
+     * @param sa The start angle for the arc, in radians.
+     * @param ea The end angle for the arc, in radians.
+     */
+    drawArc(x: number, y: number, r:number, sa:number, ea: number) {
+        this.ctx.beginPath();
+        this.ctx.arc(x, y, r, sa, ea);
+        this.ctx.stroke(); 
+    }
+
+    /**
+     * Fills a polygon at the given coorinates with the radius and number of sides specified.
+     * @param x The x coordinate for the center of the polygon.
+     * @param y The y coordinate for the center of the polygon.
+     * @param r The radius of the polygon.
+     * @param n The number of sides in the polygon
+     * @param v The rotation of the polygon in radians. Default is 0.
+     */
+    fillPolygon(x: number, y: number, r: number, n: number, v: number = 0) {

```

Comment:
I don't like default values and optionals, as it creates potential confusion for the user of the engine :)